Sources/MarathonCore/Export.swift

```diff
@@ -0,0 +1,165 @@
+/**
+ *  Marathon
+ *  Copyright (c) John Sundell 2017
+ *  Licensed under the MIT license. See LICENSE file.
+ */
+
+import Foundation
+import Files
+
+// MARK: - Error
+
+public enum ExportError {
+    case missingPath
+    case failedToExportScript(String)
+}
+
+extension ExportError: PrintableError {
+    public var message: String {
+        switch self {
+        case .missingPath:
+            return "No script name/path given"
+        case .failedToExportScript(let name):
+            return "Failed to export script \(name)"
+        }
+    }
+
+    public var hints: [String] {
+        switch self {
+        case .missingPath:
+            return ["Pass the name/path of a script file to export (for example 'marathon export myScript')"]
+        case .failedToExportScript:
+            return ["Remove directory or provide alternate export path"]
+        }
+    }
+}
+
+// MARK: - Task
+
+internal final class ExportTask: Task, Executable {
+    private typealias Error = ExportError
+
+    func execute() throws {
+
+        guard let exportScriptPath = arguments.first?.asScriptPath() else {
+            throw Error.missingPath
+        }
+
+        let script = try scriptManager.script(atPath: exportScriptPath, allowRemote: false)
+
+        let destination = try destinationFolder()
+        try overwriteIfNeeded(destination: destination, scriptName: script.name)
+
+        let tempFile = try script.folder.createFile(named: script.name.asScriptPath())
+        try tempFile.write(data: try script.folder.file(named: "OriginalFile").read())
+
+        try perform(export(tempFile, to: destination), orThrow: Error.failedToExportScript(tempFile.name))
+        try tempFile.delete()
+    }
+
+    private func destinationFolder() throws -> Folder {
+        let folder: Folder
```

Comment:
You don't need this `let`, you can just `return` directly from within the `if`/`else` blocks.